sparse_strips/vello_cpu/src/fine/mod.rs

```diff
@@ -507,20 +518,324 @@ pub(crate) mod strip {
+    /// Perform a normalized multiplication between this number and another
+    fn normalized_mul(self, other: Self) -> Self;
+    /// Extract the underlying color from a premultiplied color.
+    fn extract_color(color: &PremulColor) -> [Self; COLOR_COMPONENTS];
+    /// Convert a normalized u8 integer to this type.
+    fn from_normalized_u8(num: u8) -> Self;
+    /// Convert a plain u8 integer to this type.
+    fn from_u8(num: u8) -> Self;
+    /// Convert this number to a normalized f32.
+    fn to_f32_normalized(self) -> f32;
+    /// Convert to this number from a normalized f32.
+    fn from_normalized_f32(num: f32) -> Self;
+    // TODO: These should be sized to COLOR_COMPONENTS, but will leave that for
+    // the future.
+    /// Convert a slice to a RGBA8 slice.
+    fn to_rgba8(_in: &[Self]) -> [u8; COLOR_COMPONENTS];
+    /// Convert a RGBA8 slice to a slice of this type.
+    fn from_rgba8(_in: &[u8]) -> [Self; COLOR_COMPONENTS];
+    /// Convert a RGBAF32 slice to a slice of this type.
+    fn from_rgbaf32(_in: &[f32]) -> [Self; COLOR_COMPONENTS];
+    /// Get the inverse of the number.
+    fn inv(self) -> Self;
+    /// Get the widened representation of the current number.
+    fn widen(self) -> Self::Widened;
+}
+
+impl FineType for u8 {
+    type Widened = u16;
+    const ZERO: Self = 0;
+    const MID: Self = 127;
+    const ONE: Self = 255;
+
+    #[inline(always)]
+    fn min(self, other: Self) -> Self {
+        Ord::min(self, other)
+    }
+
+    #[inline(always)]
+    fn max(self, other: Self) -> Self {
+        Ord::max(self, other)
+    }
+
+    fn widened_mul_div(self, other: Self, other2: Self) -> Self::Widened {
+        (self.widen() * other.widen()) / other2.widen()
+    }
+
+    #[inline(always)]
+    fn normalized_mul(self, other: Self) -> Self {
+        (self.widen() * other.widen()).normalize().narrow()
+    }
+
+    #[inline(always)]
+    fn extract_color(color: &PremulColor) -> [Self; COLOR_COMPONENTS] {
+        color.as_premul_rgba8().to_u8_array()
+    }
+
+    #[inline(always)]
+    fn from_normalized_u8(num: u8) -> Self {
+        num
+    }
+
+    #[inline(always)]
+    fn from_u8(num: u8) -> Self {
+        num
+    }
+
+    #[inline(always)]
+    fn to_f32_normalized(self) -> f32 {
+        self as f32 / 255.0
+    }
+
+    #[inline(always)]
+    fn from_normalized_f32(num: f32) -> Self {
+        (num * 255.0 + 0.5) as Self
+    }
+
+    #[inline(always)]
+    fn to_rgba8(_in: &[Self]) -> [u8; COLOR_COMPONENTS] {
+        [_in[0], _in[1], _in[2], _in[3]]
+    }
+
+    #[inline(always)]
+    fn from_rgba8(_in: &[u8]) -> [Self; COLOR_COMPONENTS] {
+        [_in[0], _in[1], _in[2], _in[3]]
+    }
+
+    #[inline(always)]
+    fn from_rgbaf32(_in: &[f32]) -> [Self; COLOR_COMPONENTS] {
+        let r = |val: f32| (val * 255.0 + 0.5) as Self;
+        [r(_in[0]), r(_in[1]), r(_in[2]), r(_in[3])]
+    }
+
+    #[inline(always)]
+    fn inv(self) -> Self {
+        Self::ONE - self
+    }
```

Comment:
These can become default and be deleted here.

```suggestion
    #[inline(always)]
    fn to_u8(self) -> u8 {
        self
    }
```